orbit/pkg/bitlocker/bitlocker_management_windows.go

```diff
@@ -0,0 +1,574 @@
+	if err != nil {
+		return "", fmt.Errorf("ProtectKeyWithNumericalPassword(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return "", fmt.Errorf("ProtectKeyWithNumericalPassword(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	var recoveryKey ole.VARIANT
+	ole.VariantInit(&recoveryKey)
+	resultRaw, err = oleutil.CallMethod(v.handle, "GetKeyProtectorNumericalPassword", volumeKeyProtectorID.ToString(), &recoveryKey)
+
+	if err != nil {
+		return "", fmt.Errorf("GetKeyProtectorNumericalPassword(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return "", fmt.Errorf("GetKeyProtectorNumericalPassword(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	return recoveryKey.ToString(), nil
+}
+
+// protectWithPassphrase adds a passphrase key protector
+// https://docs.microsoft.com/en-us/windows/win32/secprov/protectkeywithpassphrase-win32-encryptablevolume
+func (v *Volume) protectWithPassphrase(passphrase string) error {
+	var volumeKeyProtectorID ole.VARIANT
+	ole.VariantInit(&volumeKeyProtectorID)
+
+	resultRaw, err := oleutil.CallMethod(v.handle, "ProtectKeyWithPassphrase", nil, passphrase, &volumeKeyProtectorID)
+	if err != nil {
+		return fmt.Errorf("protectWithPassphrase(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return fmt.Errorf("protectWithPassphrase(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	volumeKeyProtectorID.ToString()
+
+	return nil
+}
+
+// protectWithTPM adds the TPM key protector
+// https://docs.microsoft.com/en-us/windows/win32/secprov/protectkeywithtpm-win32-encryptablevolume
+func (v *Volume) protectWithTPM(platformValidationProfile *[]uint8) error {
+	var volumeKeyProtectorID ole.VARIANT
+	ole.VariantInit(&volumeKeyProtectorID)
+	var resultRaw *ole.VARIANT
+	var err error
+
+	if platformValidationProfile == nil {
+		resultRaw, err = oleutil.CallMethod(v.handle, "ProtectKeyWithTPM", nil, nil, &volumeKeyProtectorID)
+	} else {
+		resultRaw, err = oleutil.CallMethod(v.handle, "ProtectKeyWithTPM", nil, *platformValidationProfile, &volumeKeyProtectorID)
+	}
+	if err != nil {
+		return fmt.Errorf("protectKeyWithTPM(%s): %w", v.letter, err)
+	} else if val, ok := resultRaw.Value().(int32); val != 0 || !ok {
+		return fmt.Errorf("protectKeyWithTPM(%s): %w", v.letter, encryptErrHandler(val))
+	}
+
+	return nil
+}
+
+// getBitlockerStatus returns the current status of the volume
+// https://learn.microsoft.com/en-us/windows/win32/secprov/getprotectionstatus-win32-encryptablevolume
+func (v *Volume) getBitlockerStatus() (*EncryptionStatus, error) {
+	var conversionStatus int32 = 0
+	var encryptionPercentage int32 = 0
+	var encryptionFlags int32 = 0
+	var wipingStatus int32 = 0
+	var wipingPercentage int32 = 0
+	var precisionFactor int32 = 4
+	var protectionStatus int32 = 0
```

Comment:
Nit, but 0 will be automatically set, and you can group in a var ( ) block:

```suggestion
	var (
	        conversionStatus int32
                encryptionPercentage int32
                encryptionFlags int32
                wipingStatus int32
                wipingPercentage int32
                precisionFactor int32 = 4
                protectionStatus int32
        )
```

Reply:
Thanks for taking the time to explain how Go manages this! This is useful for me to keep learning the language.